Offline

jdextercsc opened the pull request:
This section checks the variable "offline"  if true, the it will get a .repo file that was created by the kragle playbook that is located in the root of the `deploy_node[0]` web server.  From there it copy that file to /etc/yum.repo.d/ to use that repo as its primary source for installing packages.

PLAY RECAP ********************************************************************
wiley-cmp-1                : ok=57   changed=39   unreachable=0    failed=0
wiley-ctlr-1               : ok=238  changed=168  unreachable=0    failed=0
wiley-ctlr-2               : ok=238  changed=169  unreachable=0    failed=0
wiley-ctlr-3               : ok=238  changed=169  unreachable=0    failed=0
wiley-scale-1              : ok=12   changed=10   unreachable=0    failed=0
wiley-scale-2              : ok=12   changed=10   unreachable=0    failed=0
wiley-scale-3              : ok=12   changed=10   unreachable=0    failed=0
wiley-swift-1              : ok=62   changed=40   unreachable=0    failed=0
wiley-swift-2              : ok=62   changed=40   unreachable=0    failed=0
wiley-swift-3              : ok=62   changed=40   unreachable=0    failed=0

[root@wiley-ctlr-1 ~]# yum repolist
Loaded plugins: product-id, subscription-manager
This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
repo id                                                       repo name                                                      status
Kragle_offline_repo                                           kragle_offline_repo                                            879
repolist: 879

jdextercsc committed:
added logic for use of offline var, instead of using subscription manager we use a .repo file for local repo on kragle

jdextercsc committed:
added logic for use of offline var, instead of using subscription manager we use a .repo file for local repo on kragle

jdextercsc committed:
changed where we get repo file from, to align with kragle offline repo creation

jdextercsc committed:
changed where we get repo file from, to align with kragle offline repo creation

jdextercsc committed:
changed where we get repo file from, to align with kragle offline repo creation

jdextercsc committed:
changed where we get repo file from, to align with kragle offline repo creation

jdextercsc committed:
adding of the offline var

jdextercsc committed:
fixed syntax and removed un-needed files

jdextercsc committed:
fixed syntax and removed un-needed files

mtnbikenc commented:
Please update this PR to reflect the changes to the kragle offline repo file location (web server).  Thanks!

jdextercsc commented on roles/subscription-manager/tasks/offline.yml:
```diff
@@ -0,0 +1,7 @@
+---
+
+- name: get repo file
```
gets repo file from url instead of releying on being updated by another playbook.

jdextercsc committed:
fixed syntax error, had to add gather_facts on kragle to know its IP address

jdextercsc commented:
We get repo file from the kragle.
TASK: [subscription-manager | get repo file] **********************************
changed: [wiley-ctlr-3]
changed: [wiley-ctlr-2]
changed: [wiley-swift-2]
changed: [wiley-ctlr-1]
changed: [wiley-swift-1]
changed: [wiley-swift-3]
changed: [wiley-cmp-1]
changed: [wiley-scale-2]
changed: [wiley-scale-3]
changed: [wiley-scale-1]

Testing failed due to missing dependency for python-keystone.  We will need to do a full run once we have the rpm list updated.

be installed\n---> Package saslwrapper.x86_64 0:0.22-5.el7 will be installed\n--> Finished Dependency Resolution\n You could try using --skip-broken to work around the problem\n You could try running: rpm -Va --nofiles --nodigest\n"]}
msg: Error: Package: python-keystone-2014.2.3-1.el7ost.noarch (Kragle_offline_repo)
           Requires: python-passlib

mtnbikenc commented on roles/subscription-manager/tasks/offline.yml:
```diff
@@ -0,0 +1,6 @@
+---
+- name: get repo file
+  get_url:
+    url: "http://{{hostvars['wiley-kragle']['ansible_default_ipv4']['address']}}/offline.repo"
```
Instead of running the get facts for deploy_node in site.yml to get the IP, could you use the `auto_deploy_node` var we define in the inventory?

jdextercsc replied:
We can do that, I was trying to make it as dynamic as possible, but I guess having the name of the node is the same kind of thing.  I'll make that change and push
good catch.

jdextercsc committed:
changed var in offline.yml to use predefined var.  removed task in site as no longer needed

mtnbikenc commented:
Code review :+1:

jcpowermac commented:
Code review :+1:
